### PR TITLE
Update calculated expression example to calculate age from date of birth

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

**Description**
Update calculated expression example to calculate age from date of birth.

See discussion https://chat.fhir.org/#narrow/stream/179266-fhirpath/topic/Calculating.20the.20period.20between.20two.20dates

The FHIRPath expression uses string manipulation to get the years of today's date and the date of birth, and calculate the age by substraction. It also compares the month and date parts of today's date and the date of birth to figure out if an additional year needs to be added to the result.

**Alternative(s) considered**
NA

**Type**
Bug fix

**Screenshots (if applicable)**

**Checklist**

- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: catalog/src/main/assets/behavior_calculated_expression.json [hunk 1]
```diff
@@ -2,48 +2,20 @@
...
-        {
-          "valueQuantity": {
-            "value": 1,
-            "unit": "months",
-            "system": "http://unitsofmeasure.org",
-            "code": "months"
+          "url": "http://hl7.org/fhir/uv/sdc/StructureDefinition/sdc-questionnaire-calculatedExpression",
+          "valueExpression": {
+            "language": "text/fhirpath",
+            "expression": "%resource.repeat(item).where(linkId='birthdate' and answer.empty().not()).select(today().toString().substring(0,4).toInteger() - answer.value.toString().substring(0,4).toInteger() + iif(today().toString().substring(5,2) & today().toString().substring(8,2) > answer.value.toString().substring(5,2) & answer.value.toString().substring(8,2), 1, 0) - 1)"
```
[the remark this answers is not in the document]
Comment (aditya-07) [reply catalog/src/main/assets/behavior_calculated_expression.json:18]: Can you add an explanation on how this expression is calculating age?

[the remark this answers is not in the document]
Comment (aditya-07) [reply catalog/src/main/assets/behavior_calculated_expression.json:18]: > Does this expression depend on the date format on the device?

Date always follow pattern [YYYY-MM-DD](https://build.fhir.org/ig/HL7/FHIRPath/#date), so I guess my question is irrelevant.

[the remark this answers is not in the document]
Comment (jingtang10) [reply catalog/src/main/assets/behavior_calculated_expression.json:18]: Yeah it's a fhir thing not platform dependent

[the remark this answers is not in the document]
Comment (jingtang10) [reply catalog/src/main/assets/behavior_calculated_expression.json:18]: > Can you add an explanation on how this expression is calculating age?

I cannot add comment to JSON - but i've added a small paragraph to this PR's description.

**Review by aditya-07 — APPROVED** [review]
LGTM 👍

**Pull request merged**